XenModel/Actions/HealthCheck/UploadServerStatusReportAction.cs

```diff
@@ -77,12 +77,16 @@ protected override void Run()
                 throw new Exception(Messages.ACTION_UPLOAD_SERVER_STATUS_REPORT_FAILED, e);
             }
 
-            if (Cancelling || Cancelled)
-                throw new CancelledException();
+            if (Cancelling || Cancelled)
+            {
+                throw new CancelledException();
+            }
+              
             else if (string.IsNullOrEmpty(uploadUuid))
             {
                 // Fail to upload the zip to CIS server.
                 log.ErrorFormat("Fail to upload the Server Status Report {0} to CIS server {1}", bundleToUpload, UPLOAD_DOMAIN_NAME);
+                Description = Messages.ACTION_UPLOAD_SERVER_STATUS_REPORT_FAILED;
```

Comment:
Setting the Description here is not needed, as throwing the exception should be enough.